build/commands/scripts/format.js

```diff
@@ -0,0 +1,132 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+const path = require('path')
+const fs = require('fs-extra')
+const prettier = require('prettier')
+const program = require('commander')
+
+const config = require('../lib/config')
+const util = require('../lib/util')
+
+program
+  .description(
+    'Format changed code with git cl format and prettier.\n' +
+      'The changed code is the difference between the current files on disk\n' +
+      'and the base branch (origin/master by default).\n' +
+      'This script is also called as part of presubmit checks.'
+  )
+  .option('--base <base branch>', 'set the destination branch for the PR')
+  .option(
+    '--full',
+    'format all lines in changed files instead of only the changed lines'
+  )
+  .option('--presubmit', 'Enables extra presubmit checks in git cl format')
+  .option(
+    '--dry-run',
+    "dry run, don't format files, just check if they are formatted"
+  )
+  .parse(process.argv)
+
+// A function that formats the code in the current diff with base branch.
+// It uses git cl format and prettier, then aggregates the results.
+const runFormat = async (options = {}) => {
+  if (!options.base) {
+    options.base = 'origin/master'
+  }
+  let cmdOptions = config.defaultOptions
+  cmdOptions.cwd = config.braveCoreDir
+  cmdOptions = util.mergeWithDefault(cmdOptions)
+  cmd = 'git'
+  args = ['cl', 'format', '--upstream=' + options.base]
+
+  args.push('--python')
+  args.push('--no-rust-fmt')
+
+  if (options.full) {
+    args.push('--full')
+  }
+
+  if (options.presubmit) {
+    args.push('--presubmit')
+  }
+
+  const errors = []
+
+  if (options.dryRun) {
+    const clFormatResult = util.run(cmd, [...args, '--dry-run'], {
+      ...cmdOptions,
+      continueOnFail: true
+    })
+    if (clFormatResult.status === 2) {
+      // format issues found. Run git cl format with --diff to print the diff
+      const diffResult = util.run(cmd, [...args, '--diff'], {
+        ...cmdOptions,
+        stdio: 'pipe'
+      })
+      errors.push(
+        diffResult.stdout.toString() + '\n' + 'git cl format checks failed'
+      )
+    } else if (clFormatResult.status !== 0) {
+      console.error('Fatal: git cl format internal error')
```

Comment:
1. let's use `Log` calls to highlight errors in the output.
2. pls output both stdout and stderr in the error message.